sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceValidator.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources
+
+import org.apache.spark.sql.catalyst.analysis.Resolver
+import org.apache.spark.sql.types.StructType
+
+
+/**
+ * Used to validate an input schema for datasource-specific cases (e.g., schema definition
+ * from an datasource option and unsupported field names). `validateSchema` is called
+ * in [[DataSource]] for the checking.
+ */
+trait DataSourceValidator {
```

Comment:
As explained above, I couldn't catch the name duplication for the JDBC case. So, I added this interface so as to catch each datasource-specific name duplication and other schema invalidation. But, I felt I went too far for the corner case...